src/trait_group.h

```diff
@@ -0,0 +1,197 @@
+#pragma once
+#ifndef TRAIT_GROUP_H
+#define TRAIT_GROUP_H
+
+#include "json.h"
+#include "mutation.h"
+#include "string.h"
+#include "string_id.h"
```

Comment:
You don't need to include "json.h" here, forward declarations of the used JSON classes are enough for the header. Probably the same for the "mutation.h". And we don't have a "string.h" header. You should include \<string\>, \<vector\> and \<memory\> here as their content is used here. Don't rely on other headers to include what your header uses.